### PR TITLE
update all swaggers to 2020-05-01 API version

### PR DESCRIPTION
**Pull request by giakas**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by allenjzhang** [comment]
Adding @JeffreyRichter for breaking change reviews.

**Review by JeffreyRichter — APPROVED** [review]

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/jobs-update.json [hunk 1]
```diff
@@ -9,24 +9,17 @@
...
         "outputs": [
           {
             "@odata.type": "#Microsoft.Media.JobOutputAsset",
-            "label": "example-custom-label",
-            "assetName": "job1 output"
+            "assetName": "job1-OutputAsset"
           }
-        ]
+        ],
+        "priority": "High"
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/jobs-update.json:22]: Did Priority move down here for a reason? Just curious why it moved from below description.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
           "tag2": "value2"
         }
       },
-      "properties": {
+      "Properties": {
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:19]: Are these case changes considered breaking in any way? Will they break any SDKs generated?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
             }
           },
           "PreviewLocator": null,
-          "streamingPolicyName": null
+          "StreamingPolicyName": null
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:41]: case change again - Breaking or not?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/streamingendpoint-update.json [hunk 4]
```diff
@@ -6,7 +6,6 @@
     "streamingEndpointName": "myStreamingEndpoint1",
     "api-version": "2020-05-01",
     "parameters": {
-      "id": null,
       "location": "West US",
       "tags": {
         "tag3": "value3",
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/streamingendpoint-update.json:11]: For all of our example tags, should we use a real example? Like "createdBy" or "Owner", or "ExampleTenantName" as an example?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 5]
```diff
@@ -5,7 +5,7 @@
   "swagger": "2.0",
   "info": {
     "title": "Azure Media Services",
-    "description": "Streaming resource management client for Azure Media Services",
+    "description": "Live streaming APIs",
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:8]: Maybe expand this a bit?  list off what resources that this API manages?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 6]
```diff
@@ -292,13 +292,60 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/mediaservices/{accountName}/liveEvents/{liveEventName}/allocate": {
+      "post": {
+        "tags": [
+          "LiveEvents"
+        ],
+        "summary": "Allocate resources for a live event",
+        "description": "A live event is in StandBy state after allocation completes, and is ready to start.",
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:301]: why upper case on StandBy ? is that the state enumeration name? 
Maybe more details on why a customer should call allocate and the benefits of this state in improving startup speed and reducing costs for stand by hot pools of live events.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 7]
```diff
@@ -401,7 +448,7 @@
           "LiveEvents"
         ],
         "summary": "Reset Live Event",
-        "description": "Resets an existing Live Event.",
+        "description": "Resets an existing live event.",
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:451]: Lets expand on this one to include what happens during a reset operation, and in what situations one should call Reset on a live event.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 8]
```diff
@@ -1080,10 +1127,10 @@
         "fragmentsPerTsSegment": {
           "format": "int32",
           "type": "integer",
-          "description": "The amount of fragments per HTTP Live Streaming (HLS) segment."
+          "description": "The number of fragments in an HTTP Live Streaming (HLS) segment in the output of the live event."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1130]: This does not impact HLS CMAF, only the legacy HLS v3 with TS segements.  We should make that clear here.  We have a gap still in that we do not provide a method for fragments per CMAF segment when using HLS.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 9]
```diff
@@ -1093,50 +1140,50 @@
...
         "assetName": {
           "type": "string",
-          "description": "The asset name."
+          "description": "The asset that the live output will write to."
         },
         "archiveWindowLength": {
           "format": "duration",
           "type": "string",
-          "description": "ISO 8601 timespan duration of the archive window length. This is duration that customer want to retain the recorded content."
+          "description": "ISO 8601 timespan duration of the archive window length. This is the maximum content length that can be archived in the asset for this live output, also the maximum content length for the rewind window."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1152]: Maybe include an ISO 8601 sample here?  Not a lot of folks know that format or are too lazy to look it up and find the part that include PTHHMMSS

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json (hunk 3 above)
Comment (JeffreyRichter) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:41]: Case changes ARE breaking

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 10]
```diff
@@ -1219,38 +1280,48 @@
...
           "enum": [
             "FragmentedMP4",
             "RTMP"
           ],
           "x-ms-enum": {
             "name": "LiveEventInputProtocol",
+            "values": [
+              {
+                "value": "FragmentedMP4",
+                "description": "Smooth streaming input will be sent by the contribution encoder to the live event."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1293]: I think it is "Smooth Streaming" caps on both.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json (hunk 2 above)
Comment (JeffreyRichter) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:19]: Yes, breaking

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 11]
```diff
@@ -1219,38 +1280,48 @@
...
         "accessControl": {
           "type": "object",
           "$ref": "#/definitions/LiveEventInputAccessControl",
-          "description": "The access control for LiveEvent Input."
+          "description": "Access control for live event input."
         },
         "keyFrameIntervalDuration": {
           "type": "string",
-          "description": "ISO 8601 timespan duration of the key frame interval duration."
+          "description": "ISO 8601 timespan duration of the key frame interval duration of the input. This value sets the EXT-X-TARGETDURATION property in the HLS output."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1310]: Add example format here.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 12]
```diff
@@ -1373,52 +1539,95 @@
...
           "$ref": "#/definitions/LiveEventEncoding",
-          "description": "The Live Event encoding."
+          "description": "Encoding settings for the live event. It configures whether a live encoder is used for the live event and settings for the live encoder if it is used."
+        },
+        "transcriptions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LiveEventTranscription"
+          },
+          "description": "Live transcription settings for the live event. See https://go.microsoft.com/fwlink/?linkid=2133742 for more information about the live transcription feature."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1564]: I like the use of FWLink to the article here.  We should look for other areas to do that in our descriptions where necessary for clarity.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 13]
```diff
@@ -1549,21 +1768,21 @@
...
         "description": {
           "type": "string",
-          "description": "The StreamingEndpoint description."
+          "description": "The streaming endpoint description."
         },
         "scaleUnits": {
           "type": "integer",
           "format": "int32",
-          "description": "The number of scale units.  Use the Scale operation to adjust this value."
+          "description": "The number of scale units. Use the Scale operation to adjust this value."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1776]: Should we link to the article on quota and also details on scaling here?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 14]
```diff
@@ -1549,21 +1768,21 @@
...
         },
         "availabilitySetName": {
           "type": "string",
-          "description": "The name of the AvailabilitySet used with this StreamingEndpoint for high availability streaming.  This value can only be set at creation time."
+          "description": "The name of the AvailabilitySet used with this streaming endpoint for high availability streaming. This value can only be set at creation time. This feature is not available everywhere, please contact Azure Support for more information."
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1780]: Is this feature still in use and available?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json [hunk 15]
```diff
@@ -1744,7 +1989,7 @@
     },
     "StreamingEndpointName": {
       "name": "streamingEndpointName",
-      "description": "The name of the StreamingEndpoint.",
+      "description": "The name of the streaming endpoint, maximum length is 24.",
```
Comment (johndeu) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1992]: Funny and strange that the event name is 32, an the endpoint name is only 24 chars in length. I'm sure that this is related to DNS for the Streaming endpoint?

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json (hunk 3 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:41]: This is in the example json. We should go with whichever case is correct.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 6 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:301]: Yes, it's using the actual enumeration state. It's unfortunate that we used incorrect casing for the enumeration and I didn't catch it in time.  Is it too late to change this now?  I think the additional information can be contained in its own separate documentation and we can add a reference to a fwlink to that in here..

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 7 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:451]: Resets an existing live event. All live outputs for the live event are deleted and the live event is stopped and started again. All assets used by the live outputs and streaming locators created on these assets are unaffected.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 8 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1130]: Shall we re-phrase to say: "The number of fragments in an HTTP Live Streaming (HLS) TS segment in the output of the live event. This setting does not impact HLS CMAF or DASH outputs."

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 9 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1152]: "Example: PT1H to indicate one hour."

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json (hunk 2 above)
Comment (giakas) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/liveevent-delete.json:19]: all of these looks like issues with examples not the actual swagger. let me work with the service team to fix these.

---
In reply to: [497809167](https://github.com/Azure/azure-rest-api-specs/pull/10960#discussion_r497809167) [](ancestors = 497809167)

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/jobs-update.json (hunk 1 above)
Comment (giakas) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/jobs-update.json:22]: these are auto generated tests.. not sure what changed in the generation process.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 15 above)
Comment (giakas) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1992]: Look like its by design.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/streamingendpoint-update.json (hunk 4 above)
Comment (giakas) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/streamingendpoint-update.json:11]: lets try to get these examples update for Nov release. I can work with Ning and you to update the examples.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 13 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1776]: I will add the link once we fix that doc page. We don't link quotas and limits for any other types of objects.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json (hunk 14 above)
Comment (ningl2) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json:1780]: Pending email discussion.

Comment (giakas) [reply specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/streamingservice.json]: ning has updated the description to add deprecated

**Pull request merged**